model/src/main/proto/performance_metrics.proto

```diff
@@ -0,0 +1,149 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Structure for an oppia metric log.
+message OppiaMetricLog {
+  // The timestamp of the event.
+  // unit: long/millis, timezone: UTC
+  int64 timestamp_millis = 1;
+
+  // The priority of the event.
+  Priority priority = 2;
+
+  // Indicates whether the application is in foreground.
+  bool is_app_in_foreground = 3;
+
+  // The memory tier of the current device.
+  MemoryTier memory_tier = 4;
+
+  // The storage tier of the current device.
+  StorageTier storage_tier = 5;
+
+  // The current network type of the device.
+  NetworkType network_type = 6;
+
+  // The current activity where the event is getting logged.
+  CurrentActivity current_activity = 7;
+
+  // Structure of the apk size package.
+  message ApkSize {
+    // Size of the android package file of the app on the device in bytes.
+    int64 apk_size_bytes = 1;
+  }
+
+  // Structure of the storage usage package.
+  message StorageUsage {
+    // Amount of storage usage by the app on user's device in bytes.
+    int64 storage_usage_bytes = 1;
+  }
+
+  // Structure of the startup latency package.
+  message StartupLatency {
+    // Amount of latency on application startup in millis.
+    int64 startup_latency_millis = 1;
+  }
+
+  // Structure of the memory usage package.
+  message MemoryUsage {
+    // Amount of memory used by the application on the device in bytes.
+    int64 total_pss_bytes = 1;
+  }
+
+  // Structure of the network usage package.
+  message NetworkUsage{
+    // Number of bytes received by the application on the device during the bucket's time interval.
+    int64 bytes_received = 1;
+
+    // Number of bytes transmitted by the application from the device during the bucket's time
+    // interval.
+    int64 bytes_transmitted = 2;
+  }
+
+  // Structure of the cpu usage package.
+  message CpuUsage{
+    // Amount of cpu used by the application.
```

Comment:
What is the unit for this field?

Reply:
CPU usage is going to be in percentages, so unit-less -- does that make sense?

Reply:
If it's a percentage, then it's presumably a ratio. What's being divided to compute it? Overall, I'm trying to dig into the context of what this value means since it's not clear from current documentation how to actually use or populate it.

Reply:
I've created a separate issue that tracks the implementation of CPU usage performance metrics. I've added a todo here to ensure correct proto message + comment goes in when we merge all our PRs.